silverback/types.py

```diff
@@ -29,6 +34,31 @@ class SilverbackStartupState(BaseModel):
     last_block_processed: int
 
 
+class CronJob(BaseModel):
+    last_run: datetime = datetime.utcnow()
+    schedule: str
+
+    @property
+    def schedule_id(self) -> str:
+        return string_4byte(self.schedule)
+
+    @property
+    def task_name(self) -> str:
+        return f"cron/{self.schedule_id}"
+
+    def mark_ran(self):
+        self.last_run = datetime.utcnow()
+
+    def should_run(self) -> bool:
+        # NOTE: Checking against ONE_MINUTE since that is the resolution of cron schedules
+        return pycron.is_now(self.schedule) and datetime.utcnow() - self.last_run >= ONE_MINUTE
+
+
+def string_4byte(s: str):
+    """Return a 4-byte hex string hash of the given string"""
+    return keccak(text=s).hex()[:8]
```

Comment:
if you have multiple cron jobs with the same time frame set, you'll have identical hex values. Do we want this? Or is this supposed to be an identifier that helps us separate which job is which?

Reply:
Silverback doesn't currently support multiple handlers.  It would raise a `DuplicateHandlerError`, same as if someone tried to use multiple new block handlers..

Reply:
Seems like this is changing in #66.  Will wait until that merges to update this PR.